### ISSUE TITLE
GitLab less permissions for the token

### ISSUE THREAD

**Issue by abitrolly** [rafaelmardojai/forge-sparks#44]
Why write `api` permissions is needed?
Would `read_api` be enough?

![image](https://github.com/user-attachments/assets/dfda3c90-bae6-4b18-bd3c-6e21cabac023)

**Comment by rafaelmardojai** [comment]
Write access is need to mark as read (actually mark the todo as done). Ideally GitLab would have a more fine grained scopes but is not the case.

**Comment by abitrolly** [comment]
@rafaelmardojai I've seen ongoing work to change how access tokens are managed, still not sure how fine grained the controls would be.

If I don't want to mark things as read from the app, and my profile is public, do I need the token at all?

**Comment by rafaelmardojai** [comment]
@abitrolly It should fail silently and the notification will stay in the app window, the app should be smart enough to not re-send a notification, but yeah, the app should work without mayor problems if you just use the `api` scope.

**Comment by gwuen** [comment]
I was wondering the same thing and stumbled upon this issue. Would it be possible to document this somewhere in the app or clarify the message in the account dialog? I'm thinking something like this:

> Forge Sparks needs the *api* (read and write) scope granted to read notifications and mark them as read.

**Comment by KlfJoat** [comment]
Yeah, I just gave it `read_api` bc I didn't want to give it full access.

Agree that this is GitLab's problem. Unfortunately, such shortcomings require adjustment by others.

I suggest the message say something like...

> Forge Sparks needs `read_api` scope to read notifications or full `api` scope to also mark todos as done.

That way, users can understand that Forge Sparks does work with read-only, and they understand the functionality they'll lose out on if they select that option.
